var/spack/repos/builtin/packages/doxygen/package.py

```diff
@@ -44,7 +44,7 @@ class Doxygen(Package):
             description='Build with dot command support from Graphviz.')
 
     depends_on("cmake@2.8.12:", type='build')
-    depends_on("flex", type='build')
+    depends_on("flex@:2.6.0", type='build')
```

Comment:
certainly NOT the case on `macOS`+`clang`, i currently have the following built:
```
^doxygen@1.8.12%clang@8.0.0-apple+graphviz arch=darwin-sierra-x86_64
        ^bison@3.0.4%clang@8.0.0-apple arch=darwin-sierra-x86_64
        ^flex@2.6.3%clang@8.0.0-apple arch=darwin-sierra-x86_64
            ^gettext@0.19.8.1%clang@8.0.0-apple+bzip2+curses+git~libunistring+libxml2+tar+xz arch=darwin-sierra-x86_64
                ^libxml2@2.9.4%clang@8.0.0-apple~python arch=darwin-sierra-x86_64
                    ^xz@5.2.3%clang@8.0.0-apple arch=darwin-sierra-x86_64
                ^tar@1.29%clang@8.0.0-apple arch=darwin-sierra-x86_64
            ^help2man@1.47.4%clang@8.0.0-apple arch=darwin-sierra-x86_64
        ^graphviz@2.38.0%clang@8.0.0-apple~perl arch=darwin-sierra-x86_64
            ^expat@2.2.0%clang@8.0.0-apple arch=darwin-sierra-x86_64
            ^freetype@2.7%clang@8.0.0-apple arch=darwin-sierra-x86_64
                ^libpng@1.6.27%clang@8.0.0-apple arch=darwin-sierra-x86_64
            ^ghostscript@9.18%clang@8.0.0-apple arch=darwin-sierra-x86_64
                ^libtiff@4.0.6%clang@8.0.0-apple arch=darwin-sierra-x86_64
                    ^jpeg@9b%clang@8.0.0-apple arch=darwin-sierra-x86_64
            ^swig@3.0.11%clang@8.0.0-apple arch=darwin-sierra-x86_64
                ^pcre@8.39%clang@8.0.0-apple+utf arch=darwin-sierra-x86_64
```
Please see whether it's platform/compiler specific or what' the error?